resolver/resolver.go

```diff
@@ -31,8 +31,9 @@ type Resolver struct {
 	rs               *records.RecordGenerator
 	rsLock           sync.RWMutex
 	rng              *rand.Rand
-	fwd              exchanger.Forwarder
 	generatorOptions []records.Option
+	zoneFwds         map[string]exchanger.Forwarder
```

Comment:
the string key is the tld for the specified forwarder, and is used later, https://github.com/mesosphere/mesos-dns/pull/441/files#diff-c6a2b1bb9d7feeb56f43abf7caadeed1R112

It's not necessary that this data structure be a map, but there's very little disadvantage to it being one IMO.